Assignments/ip1/ip1.md

```diff
@@ -0,0 +1,453 @@
+---
+layout: assignment
+title: "Individual Project 1"
+permalink: /assignments/ip1
+parent: Assignments
+nav_order: 1
+due_date: "Wednesday September 18, 2024 12:00pm EST (Noon)"
+submission_notes: Submit through Github Classroom
+---
+
+Welcome aboard to the Stack Overflow team! We're glad that you're here and ready to join our development team as a new software engineer.
+We're building an interactive application for online community to share their knowledge and experience, and are very happy to see that we have so many new developers who can help make this application a reality.
+By the end of the semester, you'll be able to propose, design, implement and test new features for our project.
+We understand that some of you may have some web development experience, but don't expect that most of you do, and hence, have created an individual project to help you get up to speed with our existing codebase and development environment.
+
+FakeStackOverFlow is a web application that consists of some code that runs in each client's web browser, and also code that runs on a server.
+
+This implementation effort will be split across two deliverables. In this first deliverable, you will implement and test the core backend components for this feature, and in the second deliverable, you will implement and test the frontend components. 
+
+## Objectives of this assignment
+The objectives of this assignment are to:
+*  Get you familiar with the basics of TypeScript, VSCode, and the project codebase
+*  Learn how to read and write code in TypeScript
+*  Translate high-level requirements into code
+*  Learn how to write unit tests with Jest
+
+## Getting started with this assignment
+
+Start by accepting our [invitation](https://classroom.github.com/a/ofMQDAQs). It will create a Github repository for you which will include the starter code for this assignment.
+
+
+### 1. Prerequisites
+
+#### 1.1 Install Node.js and npm
+
+If you haven't installed npm and node.js, follow the [tutorial on setting up a development environment for this class]({{site.baseurl}}{% link tutorials/week1-getting-started.md %}) 
+
+#### 1.2 Install MongoDB
+
+We use MongoDB as the NoSQL database to store data related to this application.  
+ 
+1. Follow the [instructions in the official MongoDB documentation](https://www.mongodb.com/docs/manual/administration/install-community/) to install the free community edition.
+2. Choose 'Install on Linux', 'Install on macOS', or 'Install on Windows', depending on your system.
+3. Scroll down to the section labeled 'Install MongoDB Community Edition.'and to [MongoDBDownload Center](https://www.mongodb.com/try/download/community?tck=docs_server).
+4. For Windows, in the Package dropdown, select `msi`.  Then download and run the installer.
+5. On Windows, select the _“Install MongoDB as a Service”_ checkbox and install.  This will start MongoDB as a background service.
+6. Install "MongoDB Compass" if prompted.
```

Comment:
For Mac, download the dmg file from https://www.mongodb.com/try/download/compass. Once the application starts:
1. Click on "Add new connection" in the left sidebar.
2. Make sure the URI field is `mongodb://localhost:27017`
3. Click on "Connect" - MongoDB will need to be running

Reply:
can u update the md file?